crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -108,6 +114,119 @@ impl IndexerStore for PgIndexerStore {
             })
     }
 
+    fn get_event(&self, id: EventID) -> Result<Event, IndexerError> {
+        let mut pg_pool_conn = get_pg_pool_connection(&self.cp)?;
+        pg_pool_conn
+            .build_transaction()
+            .read_only()
+            .run(|conn| {
+                events::table
+                    .filter(events::dsl::transaction_digest.eq(id.tx_digest.base58_encode()))
+                    .filter(events::dsl::event_sequence.eq(id.event_seq))
+                    .first::<Event>(conn)
+            })
+            .map_err(|e| {
+                IndexerError::PostgresReadError(format!(
+                    "Failed reading event in PostgresDB with error {:?}",
+                    e
+                ))
+            })
+    }
+
+    fn get_events(
+        &self,
+        query: EventQuery,
+        cursor: Option<EventID>,
+        limit: Option<usize>,
+        descending_order: bool,
+    ) -> Result<EventPage, IndexerError> {
+        let mut pg_pool_conn = get_pg_pool_connection(&self.cp)?;
+        let mut boxed_query = events::table.into_boxed();
+        match query {
+            EventQuery::All => {}
+            EventQuery::Transaction(digest) => {
+                boxed_query =
+                    boxed_query.filter(events::dsl::transaction_digest.eq(digest.base58_encode()));
+            }
+            EventQuery::MoveModule { package, module } => {
+                boxed_query = boxed_query
+                    .filter(events::dsl::package.eq(package.to_string()))
+                    .filter(events::dsl::module.eq(module));
+            }
+            EventQuery::MoveEvent(struct_name) => {
+                boxed_query = boxed_query.filter(events::dsl::event_type.eq(struct_name));
+            }
+            EventQuery::Sender(sender) => {
+                boxed_query = boxed_query.filter(events::dsl::sender.eq(sender.to_string()));
+            }
+            EventQuery::TimeRange {
+                start_time,
+                end_time,
+            } => {
+                boxed_query = boxed_query
+                    .filter(events::dsl::event_time_ms.ge(start_time as i64))
+                    .filter(events::dsl::event_time_ms.lt(end_time as i64));
+            }
+            EventQuery::EventType(_) => {}
+            _ => {
+                return Err(IndexerError::NotImplementedError(
+                    "Querying events by Recipient and Object is deprecated.".to_string(),
+                ));
+            }
+        }
+        // fetch one more item to tell if there is next page
+        let mut page_limit = limit
+            .unwrap_or(MAX_EVENT_PAGE_SIZE)
+            .min(MAX_EVENT_PAGE_SIZE)
```

Comment:
nit: I would be more explicit here by returning error if limit exceeds `MAX_EVENT_PAGE_SIZE`